### ISSUE TITLE
Always run on 1 CPU core only

### ISSUE THREAD

**Issue by minhhieugma** [vlnahp/Btcbf#39]
No matter how many cores I choose, it always runs on only one core.
It reflects on the Task Manager as well as the "current rate" metrics as well.

I guess that we need to use `Process` to achieve it?
`from multiprocessing import cpu_count, Process`

Reference: https://stackoverflow.com/questions/22700164/python-running-multiple-processes-simultaneously

**Comment by vlnahp** [comment]
Would you please let me know your cpu's specifications?

**Comment by LU5GOON** [comment]
This is interesting as i am running all 24 cores and getting 16,000 - 18,000 k/s on the BTC Puzzle

CPU used:

AMD Ryzen 9 5900X 12-Core Processor  4.20 GHz

Should i be getting more than that?

**Comment by vlnahp** [comment]
I guess it's because of using threads instead of processes. Processes don't share memory, that's why I used threads.
I have to revise this part of the program.

**Comment by minhhieugma** [comment]
Mine is i7-9750h. But I guess you know how to fix it already.

<img width="244" alt="image" src="https://user-images.githubusercontent.com/9654744/165345413-73c59cc9-421a-4008-ac9b-91a212505640.png">

**Comment by vlnahp** [comment]
Yes, I think so.
Will be fixed soon✅

**Comment by LU5GOON** [comment]
> Yes, I think so. Will be fixed soon✅

Hey @vlnahp i had a random questio of the backing of this bug report here.
Is there the possibility of the speed increasing from 15,000 k/s or is his based on specs of the machine used?
I know you mentioned threads are used at the moment but once fixed, will it be faster?

**Comment by vlnahp** [comment]
Yes it's possible. But I need some time to do some research about threading and GIL lock.

**Comment by minhhieugma** [comment]
Should I try to fix this if you are busy now?

**Comment by vlnahp** [comment]
> Should I try to fix this if you are busy now?

Yes I'm busy. I believe it is not possible to keep the `speed()` function and use multiple threads. But you can give it a try!

**Comment by minhhieugma** [comment]
Okay, let me give it a try first

**Comment by Rosario144** [comment]
Si può inserire OpenGL in modo da utilizzare anche le GPU

**Comment by Rosario144** [comment]
Taking advantage of the GPUs it will be even faster

**Comment by LU5GOON** [comment]
> Okay, let me give it a try first

Hello 👋 how did you go with this after giving it a try? Keen to know if you found a way or possibility. Thank you.

**Comment by PopeDollar** [comment]
Hello folks! Anyone found anything with balance?

**Comment by LU5GOON** [comment]
> Hello folks! Anyone found anything with balance?

Hey PopeDollar I have been using the offline brute and targeting one BTC address for about 2 months straight now via a laptop & no such luck yet.

I do believe some here have found gold 👍👍

It's all luck at the end of the day as chances are slim to none.

**Comment by telejoshi** [comment]
> Hello folks! Anyone found anything with balance?

Finding a key is like finding a grain of sand in the whole universe. You could search for millions of years without even scratching the surface ;)
https://www.reddit.com/r/CryptoCurrency/comments/w0bv37/how_secure_are_seed_phrases/

**Comment by LU5GOON** [comment]
Hi @vlnahp 

I hope all is well with your studies.
I just wanted to check in and see if yourself or anyone has found a resolution to the outstanding bug here?

I can appreciate how busy you are of course!

Lu5Goon

**Comment by vlnahp** [comment]
> Hi @vlnahp
> 
> I hope all is well with your studies. I just wanted to check in and see if yourself or anyone has found a resolution to the outstanding bug here?
> 
> I can appreciate how busy you are of course!
> 
> Lu5Goon

Hello!
Unfortunately no success yet. It's going to take time.

**Comment by telejoshi** [comment]
> No matter how many cores I choose, it always runs on only one core. It reflects on the Task Manager as well as the "current rate" metrics as well.
> 
> I guess that we need to use `Process` to achieve it? `from multiprocessing import cpu_count, Process`
> 
> Reference: https://stackoverflow.com/questions/22700164/python-running-multiple-processes-simultaneously

I made a mini version to try out _tasks_ instead of a pool:
https://github.com/telejoshi/brutithreads

You can try this one and tell if it works.

**Comment by legoxi** [comment]
me too，my 7742 only10k/s

**Comment by VJanabia** [comment]
Fortunately, I found a private key. Unfortunately, it has been empty in 2021
If you find a Bitcoin wallet with a balance, the wallet details will show up here:

4ef862ae89545a25cb75e1d56b19aef02fae6fdaea8f6cbeacf8e58e22edd480 // private key
KysDe6HB1oPnUGCuXT88Pppqu1Td9WVDzgCYes9x4B1S5aL7bd2e // hex private key in Wallet Import Format
030bdfccb1fd2aac06cec7e688f944632a8ec33871cfaedfdd08e51f462a4e9532 // public key
15x5ugXCVkzTbs24mG2bu1RkpshW3FTYW8 // P2PKH wallet address

//-------------------------------

**Comment by LU5GOON** [comment]
Hi All,

Been a while! Just wanted to check and see if this bug has been fixed or looked into? 

👍👍

**Comment by minhhieugma** [comment]
I will try to do this on the weekend or next week.

**Comment by minhhieugma** [comment]
Hey folks, I made this PR to solve this issue.

https://github.com/vlnahp/Btcbf/pull/64

**Comment by mikorist** [comment]
> Hello folks! Anyone found anything with balance?

Never....Even if you use fastecdsa and secp256k1 as ice at best it can handle 3MK/s. Which takes about a couple of billion years to find something. Even using Pollard's kangaroo algorithm *(~200 Petakeys/s per thread) takes you a few hundred years if you don't know the public key and the range the key is in. But don't worry, we like to play like children.
